### PR TITLE
add a specfile for the validateComponentTag function

### PR DESCRIPTION
**Pull request by alicewriteswrongs**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Unit tests (`npm test`) were run locally and passed
- [x] E2E Tests (`npm run test.karma.prod`) were run locally and passed
- [x] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

GitHub Issue Number: N/A

## What is the new behavior?

This just adds tests for a file that didn't have a spec file already (`src/utils/validation.ts`).

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

Basically just increasing test coverage a bit here! I came across this file while working on a different issue (#2672) and thought I might as well add some tests here.

### CODE REVIEW

File: src/utils/test/validation.spec.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import { validateComponentTag } from '../validation';
+
+describe('validation', () => {
+  it.each([' my-tag', 'my-tag ', ' my-tag '])('should error on whitespace', (tagName) => {
```
Comment (rwaskiewicz) [inline src/utils/test/validation.spec.ts:4]: Minor, can we add an additional suite/`describe` block here? I'd like to get the code to start following the general pattern of
```js
describe('filename-under-test', () => {
  describe('function-or-class-under-test', () => {
    // any other necessary suites to split up the code
      // tests
  });
});
```
(which we don't have uniformly across the codebase today).

The reason for which is that as these files grow in size/scope, it makes adding additional suites easier (in that there's less churn in adding whitespace when we need to add a new `describe` block for a new function).

```suggestion
  describe('validateComponentTag', () => {
    ... existing tests, indented by 2 spaces
```

**Review by rwaskiewicz — CHANGES_REQUESTED** [review]
All in all looks good! I had a few minor nitpicks, but nothing major.

File: src/utils/test/validation.spec.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    expect(validateComponentTag(tagName)).toBe('Tag can not contain white spaces');
+  });
+
+  it('should error on upper case', () => {
+    expect(validateComponentTag('My-Tag')).toBe('Tag can not contain upper case characters');
+  });
+
+  it('should error on non-string', () => {
```
[the remark this answers is not in the document]
Comment (alicewriteswrongs) [reply src/utils/test/validation.spec.ts:12]: yes!

**Review by rwaskiewicz — APPROVED** [review]
